cmd/aws-application-networking-k8s/main.go

```diff
@@ -72,27 +72,37 @@ func main() {
 	var metricsAddr string
 	var enableLeaderElection bool
 	var probeAddr string
+	var debug bool
 
 	// setup glog level
 	flag.Lookup("logtostderr").Value.Set("true")
-	flag.Lookup("v").Value.Set(config.GetLogLevel())
 
 	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
 	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&debug, "debug", false, "enable debug mode")
 	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
-	opts := zap.Options{
-		Development: true,
-	}
-	opts.BindFlags(flag.CommandLine)
 	flag.Parse()
 
-	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+	log := gwlog.NewLogger(debug)
+	ctrl.SetLogger(zapr.NewLogger(log.Desugar()).WithName("runtime"))
 
-	config.ConfigInit()
+	setupLog := log.Named("setup")
+	err := config.ConfigInit()
+	if err != nil {
+		setupLog.Error(err, "init config")
+		os.Exit(1)
+	}
+	setupLog.Infow("init config",
+		"VpcId", config.VpcID,
+		"Region", config.Region,
+		"AccoundId", config.AccountID,
+		"DefaultServiceNetwork", config.DefaultServiceNetwork,
+		"UseLongTgName", config.UseLongTGName,
+	)
```

Comment:
Should we log every property individually here? My concern is that, as new init configs are added, this is one more place we need to keep updating. Can we just log all of the configs here (e.g. by logging the struct)?

Reply:
Config is package name here, and all properties are package global variables. There is a ticket in the backlog to make them into a struct, but there are many places that refer to these, it's a noticeable refactoring. #310